### PR TITLE
Logging for NTDataPublisher

### PR DESCRIPTION
**Pull request by sarah-e-c**
Logging for NTDataPublisher and removal of a getInstance() method.

### CODE REVIEW

**Comment by mdurrani808** [comment]
This is pretty awesome. Any way for the pipeline logger to tell the user what pipeline is actually being set (the name) along with the index? If there was a way to get this sort of logging for resolution too, that would be much appreciated for debugging.

**Comment by sarah-e-c** [comment]
> This is pretty awesome. Any way for the pipeline logger to tell the user what pipeline is actually being set (the name) along with the index? If there was a way to get this sort of logging for resolution too, that would be much appreciated for debugging.

i can definitely try. also, im a new contributor so is there any way i can update this same PR?

**Comment by shueja-personal** [comment]
You can just keep committing to the branch you've been working on and it'll update here.

**Comment by sarah-e-c** [comment]
> You can just keep committing to the branch you've been working on and it'll update here.

it should give the nickname as well now.

**Comment by mdurrani808** [comment]
Is there a way to get logging for when a certain resolution is set? That would be one last thing, then ready for review / merge imo.

**Comment by mdurrani808** [comment]
https://github.com/PhotonVision/photonvision/blob/master/photon-core/src/main/java/org/photonvision/vision/camera/USBCameraSource.java
Relevant lines are setVideoModeInternal. There is also a setVideoModeInternal for ZeroCopyPicamSource

**Comment by pietroglyph** [comment]
>  Relevant lines are setVideoModeInternal. There is also a setVideoModeInternal for ZeroCopyPicamSource
Please do not modify each `setVideoModeInternal`. You can put the logging call in [`setVideoMode`](https://github.com/PhotonVision/photonvision/blob/b408a58e9eeaa56f29e7572a0394cbd44c1acb2e/photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceSettables.java#L64) instead—this allows you to have that call in one place instead of five, not to mention that this use case is one of the reasons for the `setVideoMode` -> `setVideoModeInternal` level of indirection in the first place.

File: photon-core/src/main/java/org/photonvision/vision/processes/PipelineManager.java [hunk 1]
```diff
@@ -92,6 +92,28 @@ public CVPipelineSettings getPipelineSettings(int index) {
...
+                    return driverModePipeline.getSettings().pipelineNickname;
+                case CAL_3D_INDEX:
+                    return calibration3dPipeline.getSettings().pipelineNickname;
+            }
+        }
+
+        for (var setting : userPipelineSettings) {
+            if (setting.pipelineIndex == index) return setting.pipelineNickname;
+        }
+        return null;
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/processes/PipelineManager.java:114]: Nit: might be better to return "", as the things that call this function don't check for null

File: photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java [hunk 2]
```diff
@@ -61,6 +66,9 @@ public class NTDataPublisher implements CVPipelineResultConsumer {
     private final Supplier<Integer> pipelineIndexSupplier;
     private final BooleanSupplier driverModeSupplier;
 
+    private final PipelineManager pipelineManager =
+            new PipelineManager(new CameraConfiguration(null, null));
+
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java:71]: Seems a bit like code leakage to maintain this state in two places? In both VisionModule and NTDataPublisher. Can the logging calls instead potentially move to the VisionModule itself? Or is there another way around this?

**Comment by sarah-e-c** [comment]
Sorry it took me so long. I just moved the logging to another place

**Review by gerth2 — APPROVED** [review]

**Pull request merged**